### PR TITLE
Load Sys GameSettings from a .zip file

### PR DESCRIPTION
**Pull request by delroth**
Currently we store default GameSettings for Dolphin in individual .ini files (1332 of them, to be precise). While this shouldn't be an issue, some operating systems (_cough_ Windows _cough_) are really slow at creating or copying large numbers of files, and GameSettings have been an issue in the past for this reason.

This PR adds support for loading Sys GameSettings from a single .zip archive. In our Git repository we still keep individual .ini files with their own commit history, etc. At build time CMake / Visual Studio are responsible for packaging the individual files into a zip archive which is then used at runtime.

Testing checklist:
- [ ] Windows
- [ ] MacOS
- [x] Linux
- [ ] Android (does not build yet)

### CODE REVIEW

File: Source/Core/Core/Core.vcxproj [hunk 1]
```diff
@@ -582,6 +582,12 @@
     <ProjectReference Include="$(ExternalsDir)pugixml\pugixml.vcxproj">
       <Project>{38fee76f-f347-484b-949c-b4649381cffb}</Project>
     </ProjectReference>
+    <ProjectReference Include="..\..\..\Data\Data.vcxproj">
```
Comment (shuffle2) [inline Source/Core/Core/Core.vcxproj]: there's probably a short var for that too?

File: Source/Core/Core/Core.vcxproj [hunk 2]
```diff
@@ -582,6 +582,12 @@
...
+      <Project>{99f1156d-29c5-4417-9d9d-23ad00099b71}</Project>
+    </ProjectReference>
+    <ProjectReference Include="..\..\..\Externals\minizip\minizip.vcxproj">
```
Comment (shuffle2) [inline Source/Core/Core/Core.vcxproj]: $(ExternalsDir)

File: Externals/minizip/minizip.vcxproj [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  <ImportGroup Label="ExtensionSettings">
+  </ImportGroup>
+  <ImportGroup Label="PropertySheets">
+    <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
+    <Import Project="..\..\Source\VSProps\Base.props" />
+    <Import Project="..\..\Source\VSProps\ClDisableAllWarnings.props" />
+  </ImportGroup>
+  <PropertyGroup Label="UserMacros" />
+  <PropertyGroup />
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'">
```
Comment (shuffle2) [inline Externals/minizip/minizip.vcxproj]: if you remove the condition, you dont need to duplicate setting `PreprocessorDefinitions`

**Comment by mbc07** [comment]
> * The "Show defaults" in game properties windows are greyed out since no gameini file is physically present to edit. We should either point it to github, or open a window with a text box showing the contents of the file. Alternatively, we could kill the button.

Probably not the cleanest way to do it but what about temporarily extracting the file then showing in the default text editor when the user press the button?

**Comment by delroth** [comment]
@mbc07 yeah, we could also do that. No opinions, I'll probably go for the easiest. I doubt this has much real world usage anyway.

**Comment by ligfx** [comment]
CMake needs a couple things to copy the zip into the generated macOS app bundle:
- get the built file location. CMake doesn't usually associate targets with locations (since that wouldn't map onto MSbuild), so an explicit path must be used: `set(zip_location "${CMAKE_BINARY_DIR}/Data/GameSettings.zip")`
- add it as a source, so it gets examined when the bundle is built: `target_sources(dolphin-emu PRIVATE ${zip_location})`
- set the MACOSX_PACKAGE_LOCATION property, so that it gets copied: `set_source_files_properties(${zip_location} PROPERTIES MACOSX_PACKAGE_LOCATION "Resources/Sys")`

**Comment by shuffle2** [comment]
This greatly slows down opening dolphin (with large gamelist/on windows) :( Haven't looked why...

File: Source/Core/Core/ConfigManager.cpp [hunk 4]
```diff
@@ -1055,28 +1077,78 @@ IniFile SConfig::LoadGameIni() const
   return LoadGameIni(GetGameID(), m_revision);
 }
 
+void SConfig::MergeDefaultGameIni(const std::string& id, std::optional<u16> revision, IniFile* out)
+{
+  for (const std::string& filename : ConfigLoaders::GetGameIniFilenames(id, revision))
+  {
+    out->Load(File::GetSysDirectory() + GAMESETTINGS_DIR DIR_SEP + filename, true);
```
Comment (JosJuice) [inline Source/Core/Core/ConfigManager.cpp]: Why does this function try to load from both the old location and from the zip? As far as I can tell, either we're not going to have any loose INIs and this line of code will just waste time, or we are going to have old loose INIs leftover (either due to the (Windows/Android only?) buildbot not being cleaned out or due to the user extracting the new Dolphin .7z file on top of an old install) and we will read stale game settings.

**Comment by delroth** [comment]
I ended up "fixing" the Show Defaults thing by removing it altogether from the properties window in DWx. It's not implemented in DQt anyway, if anyone wants to have a proper implementation there be my guest. I think it's very seldom used anyway.

**Comment by delroth** [comment]
Not going to work further on this. Feel free to pick up and reuse.

**Pull request closed without merging**